core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java

```diff
@@ -1061,8 +1063,13 @@ public NativeImageInvokerInfo build() {
             }
 
             private void handleAdditionalProperties(List<String> command) {
-                if (nativeConfig.additionalBuildArgs().isPresent()) {
-                    List<String> strings = nativeConfig.additionalBuildArgs().get();
+                Optional<List<String>>[] additionalBuildArgs = new Optional[] { nativeConfig.additionalBuildArgs(),
```

Comment:
Optional<List<String>>[] is quite the type 😆

Reply:
Yes :), even co-pilot noticed it (for different reasons though).

See "Comments suppressed due to low confidence (1)" in https://github.com/quarkusio/quarkus/pull/47883#pullrequestreview-2843476748

>